src/Microsoft.TemplateEngine.TemplateLocalizer/Commands/Export/ExportCommand.cs

```diff
@@ -0,0 +1,196 @@
+
+                if (runningExportTasks.Count == ConcurrencyLevel)
+                {
+                    // We have reached the concurrency limit. Wait for one of the tasks to finish before issuing new ones.
+                    Task<ExportResult>? completedTask = null;
+
+                    try
+                    {
+                        completedTask = await Task.WhenAny(runningExportTasks).ConfigureAwait(false);
+                        ExportResult taskResult = await completedTask.ConfigureAwait(false);
+                        exportResults.Add(taskResult);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        // Task was intentionally cancelled.
+                        Logger.LogInformation(LocalizableStrings.command_export_log_cancelled);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.LogError(LocalizableStrings.generic_log_commandExecutionFailedWithErrorMessage, e.Message);
+                        // Fatal error. Stop processing the rest of the files.
+                        cts.Cancel();
+                        failed = true;
+                        break;
+                    }
+
+                    if (completedTask != null)
+                    {
+                        runningExportTasks.Remove(completedTask);
+                    }
+                }
+            }
+
+            try
+            {
+                // Await all the remaining work.
+                await Task.WhenAll(runningExportTasks).ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                // Task.WhenAll will only throw one of the exceptions. We need to log them all. Handle this outside of catch block.
+            }
+
+            foreach (Task<ExportResult> completedTask in runningExportTasks)
+            {
+                if (completedTask.IsCanceled)
+                {
+                    continue;
+                }
+                else if (completedTask.IsFaulted)
+                {
+                    Logger.LogError(LocalizableStrings.generic_log_commandExecutionFailedWithErrorMessage, completedTask.Exception?.Flatten().InnerException?.Message);
+                }
+                else
+                {
+                    // Tasks is known to have already completed. We can get the result without await.
+                    exportResults.Add(completedTask.Result);
+                }
+            }
+
+            if (failed)
+            {
+                Logger.LogError(LocalizableStrings.generic_log_commandExecutionFailed, CommandName);
+                return 1;
+            }
+
+            PrintResults(exportResults);
+            return cts.IsCancellationRequested ? 1 : 0;
+        }
+
+        private void PrintResults(IReadOnlyList<ExportResult> results)
```

Comment:
super minor:  printing results seems like it does not belong to the export command abstraction.  if you agree, consider another pattern to carry out the printing...such as a decorator.